all_requirements.py

```diff
@@ -0,0 +1,51 @@
+import csv
+from typing import Dict
+from typing import Set
+
+from packaging.version import parse as parse_version
+
+import riotfile
+
+
+def tree_pkgs_from_riot() -> Dict[str, Set[str]]:
+
+    return _tree_pkgs_from_riot(riotfile.venv)
+
+
+def _tree_pkgs_from_riot(node: riotfile.Venv) -> Dict[str, Set]:
+    result = {pkg: _format_version_specifiers(set(versions)) for pkg, versions in node.pkgs.items()}
+    for child_venv in node.venvs:
```

Comment:
I think you want `for venv in riotfile.venv.instances():` and then `venv.pkgs`

https://github.com/DataDog/riot/blob/aa599ad0b770d293dca90ae3bd27ed57ba59be6f/riot/riot.py#L265-L292


I _think_ ?